### ISSUE TITLE
`(wrong-type-argument number-or-marker-p nil)` when "clicking" Org mode link with `embark-act`

### ISSUE THREAD

**Issue by dschrempf** [oantolin/embark#723]
Hi!

When I click a link in Org mode with `embark-act`, I get the following backtrace:

```
Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p nil)
  embark-org-target-element-context()
  #f(compiled-function (fun) #<bytecode 0x551bde5be1cc115>)(embark-org-target-element-context)
  embark--targets()
  embark-act(nil)
  funcall-interactively(embark-act nil)
  command-execute(embark-act)
```

This error only has shown up recently, but of course it may be due to my specific setup. Let me know if I can provide more details or should try with `emacs- Q`.

**Comment by idlip** [comment]
Same here with `emacs -Q`. 

I guess new org version 9.7 has some changes related to org element.

**Comment by dschrempf** [comment]
That's right! When I evaluate the `embark-org` buffer, it starts working.

However, and this is weird: When I evaluate
```
(with-eval-after-load 'org
  (require 'embark-org))
```
which is the last bit of `embark.org`, it still does NOT work. I guess this is because it loads the byte compiled file.

So I went ahead, and byte-compiled `embark-org.el`, and lo and behold the byte compiled files differ. I have no idea how to `diff` them or what's the reason for the difference... I am also using Nix, so should we report this upstream?

**Comment by oantolin** [comment]
I'm sorry, I don't think I understand what could cause this other than it seems like maybe it's some sort of installation problem? After installing Org 9.7, I reinstalled every package with an org dependency, just in case and have had no problems.

**Comment by dschrempf** [comment]
To be honest, I suspect this being a problem within Nix. For example, it could be that the Emacs used to byte-compile the file is not yet using Org mode 9.7? I will poke around further.

**Comment by idlip** [comment]
Hey @dschrempf , there is no issue with recent update.

**Comment by dschrempf** [comment]
@idlip which update are you referring to? I am still observing this problem, but I think it is unrelated to `embark`. It seems to be Nix-specific.

**Comment by idlip** [comment]
I meant to say, with emacs overlay update and its pakcages on yesterday
with org 9.7.5 there is no issue for me.

**Comment by dschrempf** [comment]
Hm, I am still observing this, even after updating `emacs-overlay.

**Comment by rmathew8-gh** [comment]
Not sure if this is a good idea, but in my case, I was able to work around the issue by doing:
```
(setq
 embark-target-finders
  (remove 'embark-org-target-element-context embark-target-finders))

```

**Comment by oantolin** [comment]
Well, that just removes most of embark's org support, @rmathew8-gh. I'd suggest instead loading org and recompiling embark-org.

**Comment by rmathew8-gh** [comment]
Hmm.. I'm using `straight`, and I don't see embark-org as a separate package. It does show in the list of features. I did a `(straight-rebuild-package "embark")` without any luck. Even with a minimal .emacs with just org and embark, I still can't seem to get `embark-act` to work on org links.

**Comment by oantolin** [comment]
Well, I already suggested it but I can suggest the same thing again: find your copy of embark-org and recompile it. More specifically, do `M-x find-library RET embark-org RET` and then `M-x elisp-byte-compile-file`. Please let me know if that works. I cannot reproduce this problem, so I suspect it is something straight.el does. I unfortunately cannot support non-standard installation methods.

**Comment by rmathew8-gh** [comment]
@oantolin I byte-compiled all files, restarted emacs, but same issue. fwiw, here's the listing

```
  /home/roy/.emacs.d/straight/repos/embark:
  drwxr-xr-x   3 roy roy   4096 Feb 23 22:04 .
  -rw-r--r--   1 roy roy    144 Feb 23 22:04 .gitignore.elc
  -rw-r--r--   1 roy roy  27462 Feb 23 22:04 embark-org.elc
  -rw-r--r--   1 roy roy 163718 Feb 23 22:04 embark.elc
  -rw-r--r--   1 roy roy  15235 Feb 23 22:04 embark-consult.elc
  -rw-r--r--   1 roy roy     87 Feb 23 22:04 .elpaignore.elc
  -rw-r--r--   1 roy roy    158 Feb 23 22:04 .dir-locals.elc
  drwxr-xr-x   7 roy roy   4096 Feb 23 22:03 .git
  drwxr-xr-x 109 roy roy   4096 Feb 12 10:07 ..
  -rw-r--r--   1 roy roy  19657 Feb  5 07:28 embark-consult.el
  -rw-r--r--   1 roy roy   3390 Jan 27 13:10 avy-embark-collect.el
  -rw-r--r--   1 roy roy 191323 Jan 27 13:10 embark.el
  -rw-r--r--   1 roy roy  26769 Jan 27 13:10 embark-org.el
  -rw-r--r--   1 roy roy  68992 Jan 27 13:10 embark.texi
  -rw-r--r--   1 roy roy  35149 Jan 27 13:10 LICENSE
  -rw-r--r--   1 roy roy  63156 Jan 27 13:10 README.org
  -rw-r--r--   1 roy roy   5627 Jan 27 13:10 CHANGELOG.org
  -rw-r--r--   1 roy roy    177 Jan 27 13:10 .dir-locals.el
  -rw-r--r--   1 roy roy      7 Jan 27 13:10 .elpaignore
  -rw-r--r--   1 roy roy     30 Jan 27 13:10 .gitignore
```
I'm using commit `d5df0eff182b014ab49328a4dbb1d69eb7faafbd`

**Comment by oantolin** [comment]
Can you reproduce the problem from `emacs -Q`?

**Comment by hrehfeld** [comment]
I also get this (not with emacs -Q). When I edebug-instrument `embark-org-target-element-context`, everything seems to work fine afterwards 😱

**Comment by rmathew8-gh** [comment]
> Can you reproduce the problem from `emacs -Q`?

Hmm.. just switched to emacs-30 and the issue seems to be fixed!

**Comment by oantolin** [comment]
That's good news! But very mysterious...

**Comment by oantolin** [comment]
I think this issue can safely be closed.
